### PR TITLE
Optimize API request time:

### PR DESCRIPTION
**Pull request by PJLewandowski**
- add mongodb to store data instead of indexing blockchain on every
restart
- add daemon to poll blockchain changes and fill mongo database (API
should only query data)

### CODE REVIEW

File: src/blockchain/rpc.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import requests
+import json
+import random
+
+url = "http://user:password@bitcoind:8332"
+
+
+def execute_rpc(rpc):
```
Comment (mtwaro) [inline src/blockchain/rpc.py]:
```suggestion
def execute_rpc(rpc):
```
I would move this method to RPC class as "RPC.execute"

File: src/db/db.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        tx_key = txid + str(out_n)
+        db_utxo = _create_utxo(tx_key, out_addr, out.value)
+        if not utxo(id=db_utxo.id):
+            _set_insert_or_update_time(db_utxo)
+            db_utxo.save(force_insert=True)
+            new_utxo[db_utxo.id] = db_utxo
+            print('Added utxo: ' + db_utxo.to_json())
+        
+def save_address(address):
+    qs = Address.objects(hash=address.hash)
```
Comment (mtwaro) [inline src/db/db.py]: More descriptive variable name

File: src/db/db.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        return
+    
+    json_block = json.dumps(block)
+    db_block = Block.from_json(json_block)
+    _set_insert_or_update_time(db_block)
+    db_block.save(force_insert=True)
+    print('Added block(' + str(db_block.height) + ') to database')
+    utxos = {}
+    _after_insert_block(db_block, utxos)
+    return utxos
```
Comment (mtwaro) [inline src/db/db.py]: It's a bad programming practice when a "save_block" function all of a sudden returns utxos.

File: src/blockchain/rpc.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+def save_json(in_dict, name):
+    dict_json = json.dumps(in_dict)
+    with open(name, "w") as f:
+        f.write(dict_json)
+
+
+def load_file(name):
+    with open(name, 'r') as f:
+        res = json.load(f)
+    return res
```
Comment (mtwaro) [inline src/blockchain/rpc.py]: Unnecessary

File: src/db/db.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        try: 
+            db_utxo = utxo.get(id=tx_key)
+            if db_utxo.id in new_utxo:
+                del new_utxo[db_utxo.id]
+            db_utxo.delete()
+            print('Deleted utxo: ' + db_utxo.to_json())
+        except:
+            exception_string = "Invalid input. txid: " + str(txid) + ", vin_txid: " + str(
+                in_txid) + ", vout: " + str(in_vout)
+            raise BaseException(exception_string)
```
Comment (mtwaro) [inline src/db/db.py:50]: This exception needs to be handled

File: src/daemon/daemon.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    def _stall_if_running_by_docker(self, pid):
+        if self.capacity == 'docker':
+            while True:
+                try:
+                    os.kill(pid, 0)
+                except OSError:
+                    break
+        sys.exit(0)
+        
+    def daemonize(self):
```
Comment (mtwaro) [inline src/daemon/daemon.py]: Daemonization seems to be unnecessary when running inside a docker container

**Review by mtwaro — CHANGES_REQUESTED** [review]

File: src/blockchain/rpc.py (hunk 1 above)
Comment (PJLewandowski) [reply src/blockchain/rpc.py]: Done

File: src/db/db.py (hunk 2 above)
Comment (PJLewandowski) [reply src/db/db.py]: Done

File: src/db/db.py (hunk 3 above)
Comment (PJLewandowski) [reply src/db/db.py]: Done

File: src/blockchain/rpc.py (hunk 4 above)
Comment (PJLewandowski) [reply src/blockchain/rpc.py]: Done

File: src/db/db.py (hunk 5 above)
Comment (PJLewandowski) [reply src/db/db.py:50]: Fixed by workaround which resets DB on every daemon restart.

File: src/daemon/daemon.py (hunk 6 above)
Comment (PJLewandowski) [reply src/daemon/daemon.py]: Removed demonization.

**Pull request merged**